apiserver/facades/client/application/application.go

```diff
@@ -2712,14 +2723,14 @@ func (api *APIBase) unitResultForUnit(ctx context.Context, unit Unit) (*params.U
 // specified unit and machine arguments without distinguishing between port
 // ranges across subnets. This method is provided for backwards compatibility
 // with pre 2.9 agents which assume open-ports apply to all subnets.
-func (api *APIBase) openPortsOnMachineForUnit(unitName, machineID string) ([]string, error) {
+func (api *APIBase) openPortsOnUnit(ctx context.Context, unitUUID coreunit.UUID) ([]string, error) {
 	var result []string
-	machinePortRanges, err := api.model.OpenedPortRangesForMachine(machineID)
+
+	groupedPortRanges, err := api.portService.GetUnitOpenedPorts(ctx, unitUUID)
```

Comment:
It might be the only call site at the moment, but it likely will not always be. Future call sites will require ports to be broken down by endpoint

e.g.
https://github.com/juju/juju/blob/main/apiserver/facades/agent/uniter/uniter.go#L212